### PR TITLE
86cufjjjg/Multilingual-support

### PR DESCRIPTION
**Pull request by seraGeorge**
### Summary:
This pull request introduces a enhancement by implementing language state management in the store. The changes involve the creating a dedicated slice, and configuring the reducer using the reduxjs toolkit.

### Changes:

**1. Language State Setup :**

- Introduced a language state with the currentLang property, utilizing the enum values 'en' (English) and 'es' (Spanish). 

**2. Language Switching :**

- Implemented the ability to switch languages by selecting the corresponding item from the language menu. The selected language is then stored in the state, ensuring a seamless and dynamic user experience.

### Additional Notes
Please review the changes and feel free to provide suggestions for improvement or flag any identified issues. Your insights are crucial to ensuring the robustness and effectiveness of the language management feature.

### CODE REVIEW

File: src/app/features/languageSlice.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { ILanguage } from '../../config';
+import i18n from '../../i18n/i18n';
+
+interface LanguageState {
```
Comment (akshay-krishna) [inline src/app/features/languageSlice.ts]: If you are naming an interface, it is better to name it starting with I;
Ex: ILanguageState.

File: src/app/features/languageSlice.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const languageSlice = createSlice({
+    name: 'i18n',
+    initialState,
+    reducers: {
+        toggleLanguage: (state, action: PayloadAction<ILanguage>) => {
+            state.currentLanguage = action.payload;
+            i18n.changeLanguage(action.payload)
+                .then(() => {})
+                .catch((error) => {
+                    console.error('Something went wrong loading', error);
```
Comment (sania-salim) [inline src/app/features/languageSlice.ts]: Could avoid logging errors to console, mui Snackbar may be used instead.

File: src/app/features/languageSlice.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+};
+
+const languageSlice = createSlice({
+    name: 'i18n',
+    initialState,
+    reducers: {
+        toggleLanguage: (state, action: PayloadAction<ILanguage>) => {
+            state.currentLanguage = action.payload;
+            i18n.changeLanguage(action.payload)
+                .then(() => {})
```
Comment (sania-salim) [inline src/app/features/languageSlice.ts]: Is the .then() portion left empty for later use? Could add a comment for clarity.

File: src/app/features/languageSlice.ts (hunk 2 above)
Comment (seraGeorge) [reply src/app/features/languageSlice.ts]: Will resolve once we make a custom toast.

File: src/app/features/languageSlice.ts (hunk 1 above)
Comment (seraGeorge) [reply src/app/features/languageSlice.ts]: Resolved in a92d1ee

File: src/app/features/languageSlice.ts (hunk 3 above)
Comment (seraGeorge) [reply src/app/features/languageSlice.ts]: Resolved in b11ffac

**Review by akshay-krishna — APPROVED** [review]

File: src/app/features/languageSlice.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import i18n from '../../i18n/i18n';
+import { eLanguage } from '../../interfaces/index';
+
+interface ILanguageState {
+    currentLanguage: eLanguage;
```
Comment (lekshmi615213) [inline src/app/features/languageSlice.ts]: may besome other meaningful name than elanguage?

File: src/app/features/languageSlice.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
```
Comment (lekshmi615213) [inline src/app/features/languageSlice.ts:1]: changelog?

File: src/app/features/languageSlice.ts (hunk 4 above)
Comment (seraGeorge) [reply src/app/features/languageSlice.ts]: Resolved in 68d3167516aa920351ce775de2d84003ce48c784

File: src/app/features/languageSlice.ts (hunk 5 above)
Comment (seraGeorge) [reply src/app/features/languageSlice.ts:1]: Resolved in b4538f50cff6b58a39235132e5da1145b29ce061

**Review by sania-salim — APPROVED** [review]

**Pull request merged**